### PR TITLE
[JAX] Fix JAX distributed unit tests

### PR DESCRIPTION
**Pull request by zlsh80826**
The current qa scripts run `tests/jax/test_distributed_custom_ops.py` but there is no this file. This PR fixes this issue and also remove an assertion of `NO_MASK`.

### CODE REVIEW

**Review by denera — APPROVED** [review]

File: transformer_engine/jax/fused_attn.py [hunk 1]
```diff
@@ -54,9 +54,6 @@ def self_fused_attn(qkv: jnp.ndarray, bias: jnp.ndarray, mask: jnp.ndarray, seed
     """
     Self fused attention wrapper
     """
-    assert attn_mask_type is not AttnMaskType.NO_MASK, \
-        "Currently not support AttnMaskType.NO_MASK."
```
[the remark this answers is not in the document]
Comment (zlsh80826) [reply transformer_engine/jax/fused_attn.py:58]: 1. The fused attention custom calls can support the `NO_MASK`, but we haven't support `NO_MASK` in the TransformerLayer/MultiheadAttention high level API.
2. `CAUSAL_MASK` means to do the full cascal masking, it is not the real case for both PAXML and T5X, their inputs are indeed the padded token (That means PAXML uses the wrong implementation for several months, though it didn't affect the training). The detailed discussion is [here](https://github.com/NVIDIA/TransformerEngine/pull/469). If people encounts the no kernel issue with `PADDING_CAUSAL_MASK`, please suggest them to use `cudnn >= 8.9.6`.

**Review by mingxu1067 — APPROVED** [review]
LGTM
